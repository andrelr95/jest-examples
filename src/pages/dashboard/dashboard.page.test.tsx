import React from 'react'
import { create } from 'react-test-renderer'
import { render, screen, cleanup } from '@testing-library/react'

import DashboardPage from './dashboard.page'

const responseMock = [
  {
    id: 1,
    title: "Post 1 TEST MOCK"
  },
  {
    id: 2,
    "title": "Post 2 TEST MOCK"
  },
  {
    id: 3,
    title: "Post 3 TEST MOCK"
  }
]

beforeEach(() => {
  jest.spyOn(global, 'fetch').mockImplementationOnce(() => {
    return new Promise<Response>((resolve) => {
      return resolve({
        ...new Response(),
        json: () => Promise.resolve([...responseMock])
      })
    })
  })
})

afterEach(cleanup)

describe('<DashboardPage /> snapshot test', () => {
  it('should match snapshot', () => {
    const component = create(<DashboardPage />).toJSON()
    expect(component).toMatchSnapshot()
  })
})

describe('<DashboardPage /> unit tests', () => {
  it('should exists', () => {
    expect(DashboardPage).toBeDefined()
  })
  it('should render all posts', async () => {
    // Given
    const spy = jest.spyOn(DashboardPage.prototype, 'componentDidMount');
    render(<DashboardPage />)
    
    //Then
    expect(spy).toHaveBeenCalled()
    expect(screen.getByText('...loading page')).toBeInTheDocument()
    const renderedPosts = await screen.findAllByTestId('card-post-test')
    expect(renderedPosts).toHaveLength(3)
  })
})