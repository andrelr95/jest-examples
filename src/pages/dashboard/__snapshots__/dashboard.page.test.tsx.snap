// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardPage /> snapshot test should match snapshot 1`] = `
<span>
  ...loading page
</span>
`;
